File: PE41/PE41.py
def checkpan(n):
    nstr = str(n)
    ndigits = len(nstr)
    for i in range(1, ndigits+1):
        if str(i) not in nstr:
            return False
    return True

def checkprime(n):
    if (n < 2):
        return False
    if (n < 4):
        return True
    for i in range(2, int(n**0.5)+1):
        if (n % i == 0):
            return False
    return True

mymax = 0
for i in range(1, 10000000):
    if (checkpan(i) and checkprime(i)):
        mymax = i
print(mymax)

File: PE41/PE41.java
package projecteuler41;

public class PE41 {
	static boolean checkpan(int n) {
		String nstr = Integer.toString(n);
		int ndigits = nstr.length();
		for (int i = 1; i < ndigits + 1; i++) {
			if (!nstr.contains(Integer.toString(i))) {
				return false;
			}
		}
		return true;
	}
	
	static boolean checkprime(int n) {
		if (n < 2) return false;
		if (n < 4) return true;
		for (int i = 2; i < (int)Math.pow(n, 0.5)+1; i++) {
			if (n % i == 0) return false;
		}
		return true;
	}
	
	public static void main(String args[]) {
		int mymax = 0;
		for (int i = 1; i < 100000000; i++) {
			if (checkpan(i) && checkprime(i)) {
				mymax = i;
			}
		}
		System.out.println(mymax);
	}
}
